packages/ddp-client/livedata_connection.js

```diff
@@ -1490,6 +1541,11 @@ _.extend(Connection.prototype, {
 
     var self = this;
 
+    // Lets make sure there are no buffered writes before returning result.
+    if (!_.isEmpty(self._bufferedWrites)) {
```

Comment:
Not sure about this one; 

I added the `_.isEmpty` check here, as every call to `_flushBufferedWrites` will call `_runAfterUpdateCallbacks` (even if empty). So flushing without this predicate would mean that the after-update callbacks would run on every method result.

Alternatively the call to `_runAfterUpdateCallbacks` could be predicated / hoisted out to another function.